lib/bolt/fiber_executor.rb

```diff
@@ -78,19 +82,19 @@ def round_robin
 
         # If the future errored and the main plan has already exited, log the
         # error at warn level.
-        unless plan_futures.map(&:id).include?(0) || future.state == "done"
+        unless active_futures.map(&:id).include?(0) || future.state == "done"
           Bolt::Logger.warn('errored_futures', "Error in future '#{future.name}': #{future.value}")
         end
 
         # Remove the PlanFuture from the FiberExecutor.
-        plan_futures.delete(future)
+        finished_futures.push(active_futures.delete(future))
```

Comment:
A little confused. What is being returned by `active_futures.delete(future)` ? Is this a more concise way of deleting the future from `active_futures` in one line and adding the same future to `finished_futures` in another?

Reply:
Yeah, here `active_futures` is an Array, and [Ruby's Array#delete method](https://ruby-doc.org/core-2.6.2/Array.html#method-i-delete) removes the element(s) from the Array and returns the element.